src/pages/index.js

```diff
@@ -5,72 +5,36 @@ import Section from "../components/Section.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, settings } from "../utils/constants.js";
-
-/* Profile Var */
-const profileEditBtn = document.querySelector("#profile-edit-btn");
-const profileTitleInput = document.querySelector("#profile-title-input");
-const profileEditForm = document.forms["profile-form"];
-const profileDescriptionInput = document.querySelector(
-  "#profile-description-input"
-);
+import { settings, elements } from "../utils/constants.js";
+import Api from "../components/Api.js";
 
-/* Places Var */
-const placesAddBtn = document.querySelector("#places-add-btn");
-const placeAddForm = document.forms["add-place-form"];
+const {
+  profileEditBtn,
+  profileTitleInput,
+  profileEditForm,
+  profileDescriptionInput,
+  placesAddBtn,
+  placeAddForm,
+  avatarUpdateBtn,
+  avatarForm,
+} = elements;
 
-/* Edit Profile Button Listener */
-profileEditBtn.addEventListener("click", () => {
-  const { name, job } = userInfo.getUserInfo();
-  profileTitleInput.value = name;
-  profileDescriptionInput.value = job;
-  editProfileModal.open();
-  profileEditValidation.resetValidation();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "452c8a3d-4805-4b3d-97d0-6df5b94e48de",
+    "content-type": "application/json",
+  },
 });
 
-/* Profile Edit Function */
-function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo(formData);
-  editProfileModal.close();
-}
-
-/* Add Place Button Listener */
-placesAddBtn.addEventListener("click", () => {
-  addPlaceModal.open();
-});
-
-/* Add Place Function */
-function handleNewPlaceSubmit(cardData) {
-  const { title, url } = cardData;
-  const cardElement = getCardElement({ name: title, link: url });
-  section.addItem(cardElement);
-  addPlaceModal.close();
-  addPlaceModal.reset();
-  addPlaceValidation.disableButton();
-}
-
 /* Image Click Preview Function */
 function handleImageClick(imageData) {
   imagePreviewModal.open(imageData);
 }
 
-function getCardElement(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  const cardElement = card.getView();
-  return cardElement;
-}
-
 const profileEditValidation = new FormValidator(settings, profileEditForm);
 const addPlaceValidation = new FormValidator(settings, placeAddForm);
-
-const section = new Section(
-  { items: initialCards, renderer: getCardElement },
-  ".cards__list"
-);
-
-const imagePreviewModal = new PopupWithImage("#places-preview-modal");
-
-const userInfo = new UserInfo();
+const avatarValidation = new FormValidator(settings, avatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```